src/data/int/basic.lean

```diff
@@ -351,6 +351,35 @@ begin
   cases hk; exact ⟨_, hk⟩
 end
 
+lemma nat_abs_eq_nat_abs_iff_of_nonneg_of_nonneg {a b : ℤ} (ha : 0 ≤ a) (hb : 0 ≤ b) :
+  nat_abs a = nat_abs b ↔ a = b :=
+by rw [←sq_eq_sq ha hb, ←nat_abs_eq_iff_sq_eq]
+
+lemma nat_abs_eq_nat_abs_iff_of_nonpos_of_nonpos {a b : ℤ} (ha : a ≤ 0) (hb : b ≤ 0) :
+  nat_abs a = nat_abs b ↔ a = b :=
+begin
+  have := nat_abs_eq_nat_abs_iff_of_nonneg_of_nonneg
+    (neg_nonneg_of_nonpos ha) (neg_nonneg_of_nonpos hb),
+  simp only [int.nat_abs_neg, neg_inj] at this,
+  exact this,
+end
+
+lemma nat_abs_eq_nat_abs_iff_of_nonneg_of_nonpos {a b : ℤ} (ha : 0 ≤ a) (hb : b ≤ 0) :
+  nat_abs a = nat_abs b ↔ a = -b :=
+begin
+  have := nat_abs_eq_nat_abs_iff_of_nonneg_of_nonneg ha (neg_nonneg_of_nonpos hb),
+  simp only [int.nat_abs_neg] at this,
+  exact this,
+end
```

Comment:
Untested:
```suggestion
by simpa only [int.nat_abs_neg] using nat_abs_eq_nat_abs_iff_of_nonneg_of_nonneg ha (neg_nonneg_of_nonpos hb)
```
(might also need line wrapping)

Reply:
Thanks, I've shortened that proof and the two others